### ISSUE TITLE
LoadError (cannot load such file -- jenkins_api_client)

### ISSUE THREAD

**Issue by n-rodriguez** [jbox-web/redmine_jenkins#28]
I get this error after installing plugin into bitnami redmine distribution 2.6.1-1, here's my log entry:

```
Started GET "/redmine/projects/opt-bundle_production/jenkins_settings/test_connection" for 10.101.120.30 at 2015-01-28 10:30:20 -0600
Processing by JenkinsSettingsController#test_connection as */*
  Parameters: {"project_id"=>"opt-bundle_production"}
  Current user: devadmin (id=1)
Completed 500 Internal Server Error in 6.1ms

LoadError (cannot load such file -- jenkins_api_client):
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `block in require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  plugins/redmine_jenkins/app/services/jenkins_client.rb:1:in `<top (required)>'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `block in require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:359:in `require_or_load'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:502:in `load_missing_constant'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:192:in `block in const_missing'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `each'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `const_missing'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:514:in `load_missing_constant'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:192:in `block in const_missing'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `each'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `const_missing'
  plugins/redmine_jenkins/app/models/jenkins_setting.rb:37:in `jenkins_client'
  plugins/redmine_jenkins/app/models/jenkins_setting.rb:19:in `test_connection'
  plugins/redmine_jenkins/app/controllers/jenkins_settings_controller.rb:34:in `test_connection'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/base.rb:167:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/rendering.rb:10:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/callbacks.rb:18:in `block in process_action'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:491:in `_run__4231642489893389810__process_action__1609125497180776501__callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in `__run_callback'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:385:in `_run_process_action_callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:81:in `run_callbacks'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/callbacks.rb:17:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/rescue.rb:29:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/instrumentation.rb:30:in `block in process_action'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/notifications.rb:123:in `block in instrument'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/notifications.rb:123:in `instrument'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/instrumentation.rb:29:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/params_wrapper.rb:207:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/activerecord-3.2.21/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/base.rb:121:in `process'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/rendering.rb:45:in `process'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal.rb:203:in `dispatch'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/rack_delegation.rb:14:in `dispatch'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal.rb:246:in `block in action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:73:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:73:in `dispatch'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:36:in `call'
  vendor/bundle/ruby/2.0.0/gems/journey-1.0.4/lib/journey/router.rb:68:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/journey-1.0.4/lib/journey/router.rb:56:in `each'
  vendor/bundle/ruby/2.0.0/gems/journey-1.0.4/lib/journey/router.rb:56:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:608:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-openid-1.4.1/lib/rack/openid.rb:99:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:608:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-openid-1.4.1/lib/rack/openid.rb:99:in `call'
  vendor/bundle/ruby/2.0.0/gems/request_store-1.0.5/lib/request_store/middleware.rb:9:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/best_standards_support.rb:17:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/etag.rb:23:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/conditionalget.rb:25:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/head.rb:14:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/params_parser.rb:21:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/flash.rb:242:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/session/abstract/id.rb:210:in `context'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/session/abstract/id.rb:205:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/cookies.rb:341:in `call'
  vendor/bundle/ruby/2.0.0/gems/activerecord-3.2.21/lib/active_record/query_cache.rb:64:in `call'
  vendor/bundle/ruby/2.0.0/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:479:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in `_run__2829293897372425608__call__1188188910101496906__callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in `__run_callback'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:385:in `_run_call_callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:81:in `run_callbacks'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/remote_ip.rb:31:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/debug_exceptions.rb:16:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/show_exceptions.rb:56:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/rack/logger.rb:32:in `call_app'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/rack/logger.rb:16:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/tagged_logging.rb:22:in `tagged'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/rack/logger.rb:16:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/request_id.rb:22:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/methodoverride.rb:21:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/runtime.rb:17:in `call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/cache/strategy/local_cache.rb:72:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/lock.rb:15:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/static.rb:83:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:136:in `forward'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:245:in `fetch'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:185:in `lookup'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:66:in `call!'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:51:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/engine.rb:484:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/application.rb:231:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/railtie/configurable.rb:30:in `method_missing'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/builder.rb:134:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/urlmap.rb:64:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/urlmap.rb:49:in `each'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/urlmap.rb:49:in `call'
  passenger (4.0.57) lib/phusion_passenger/rack/thread_handler_extension.rb:74:in `process_request'
  passenger (4.0.57) lib/phusion_passenger/request_handler/thread_handler.rb:141:in `accept_and_process_next_request'
  passenger (4.0.57) lib/phusion_passenger/request_handler/thread_handler.rb:109:in `main_loop'
  passenger (4.0.57) lib/phusion_passenger/request_handler.rb:455:in `block (3 levels) in start_threads'
```

**Comment by n-rodriguez** [comment]
@webfractor did you run `bundle install --deployment`?

**Comment by webfractor** [comment]
nope, I just ran, bundle install

I went and ran: bundle install --deployment
then RAILS_ENV=production rake redmine:plugins:migrate <again>

those appears to run successfully, still cannot get anything back from test or add a job in the settings.

most current log message:

```
Started GET "/redmine/projects/opt-bundle_production/jenkins_settings/test_connection" for 10.101.120.30 at 2015-01-28 11:11:31 -0600
Processing by JenkinsSettingsController#test_connection as */*
  Parameters: {"project_id"=>"opt-bundle_production"}
  Current user: devadmin (id=1)
Completed 500 Internal Server Error in 7.5ms

LoadError (cannot load such file -- jenkins_api_client):
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `block in require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  plugins/redmine_jenkins/app/services/jenkins_client.rb:1:in `<top (required)>'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `block in require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:236:in `load_dependency'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:251:in `require'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:359:in `require_or_load'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:502:in `load_missing_constant'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:192:in `block in const_missing'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `each'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `const_missing'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:514:in `load_missing_constant'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:192:in `block in const_missing'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `each'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/dependencies.rb:190:in `const_missing'
  plugins/redmine_jenkins/app/models/jenkins_setting.rb:37:in `jenkins_client'
  plugins/redmine_jenkins/app/models/jenkins_setting.rb:19:in `test_connection'
  plugins/redmine_jenkins/app/controllers/jenkins_settings_controller.rb:34:in `test_connection'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/base.rb:167:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/rendering.rb:10:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/callbacks.rb:18:in `block in process_action'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:491:in `_run__4140739262213794261__process_action__569075715484980832__callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in `__run_callback'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:385:in `_run_process_action_callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:81:in `run_callbacks'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/callbacks.rb:17:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/rescue.rb:29:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/instrumentation.rb:30:in `block in process_action'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/notifications.rb:123:in `block in instrument'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/notifications.rb:123:in `instrument'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/instrumentation.rb:29:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/params_wrapper.rb:207:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/activerecord-3.2.21/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/base.rb:121:in `process'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/abstract_controller/rendering.rb:45:in `process'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal.rb:203:in `dispatch'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal/rack_delegation.rb:14:in `dispatch'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_controller/metal.rb:246:in `block in action'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:73:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:73:in `dispatch'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:36:in `call'
  vendor/bundle/ruby/2.0.0/gems/journey-1.0.4/lib/journey/router.rb:68:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/journey-1.0.4/lib/journey/router.rb:56:in `each'
  vendor/bundle/ruby/2.0.0/gems/journey-1.0.4/lib/journey/router.rb:56:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/routing/route_set.rb:608:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-openid-1.4.1/lib/rack/openid.rb:99:in `call'
  vendor/bundle/ruby/2.0.0/gems/request_store-1.0.5/lib/request_store/middleware.rb:9:in `call'
   vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/best_standards_support.rb:17:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/etag.rb:23:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/conditionalget.rb:25:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/head.rb:14:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/params_parser.rb:21:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/flash.rb:242:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/session/abstract/id.rb:210:in `context'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/session/abstract/id.rb:205:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/cookies.rb:341:in `call'
  vendor/bundle/ruby/2.0.0/gems/activerecord-3.2.21/lib/active_record/query_cache.rb:64:in `call'
  vendor/bundle/ruby/2.0.0/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:479:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in `_run__3648229554760281025__call__2782033802967861593__callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:405:in `__run_callback'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:385:in `_run_call_callbacks'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/callbacks.rb:81:in `run_callbacks'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/remote_ip.rb:31:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/debug_exceptions.rb:16:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/show_exceptions.rb:56:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/rack/logger.rb:32:in `call_app'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/rack/logger.rb:16:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/tagged_logging.rb:22:in `tagged'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/rack/logger.rb:16:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/request_id.rb:22:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/methodoverride.rb:21:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/runtime.rb:17:in `call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-3.2.21/lib/active_support/cache/strategy/local_cache.rb:72:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/lock.rb:15:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.21/lib/action_dispatch/middleware/static.rb:83:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:136:in `forward'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:245:in `fetch'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:185:in `lookup'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:66:in `call!'
  vendor/bundle/ruby/2.0.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:51:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/engine.rb:484:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/application.rb:231:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-3.2.21/lib/rails/railtie/configurable.rb:30:in `method_missing'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/builder.rb:134:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/urlmap.rb:64:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/urlmap.rb:49:in `each'
  vendor/bundle/ruby/2.0.0/gems/rack-1.4.5/lib/rack/urlmap.rb:49:in `call'
  passenger (4.0.57) lib/phusion_passenger/rack/thread_handler_extension.rb:74:in `process_request'
  passenger (4.0.57) lib/phusion_passenger/request_handler/thread_handler.rb:141:in `accept_and_process_next_request'
  passenger (4.0.57) lib/phusion_passenger/request_handler/thread_handler.rb:109:in `main_loop'
  passenger (4.0.57) lib/phusion_passenger/request_handler.rb:455:in `block (3 levels) in start_threads'
```

**Comment by n-rodriguez** [comment]
Hi there!

Can you please test with the latest devel branch commit?

**Comment by webfractor** [comment]
I have started from scratch as I needed to get some other issue trackers installed and integration with jenkins is a must. As such, I have different version numbers now, here is the full process, in order:

```
sudo -i 
wget https://downloads.bitnami.com/files/stacks/redmine/2.6.2-0/bitnami-redmine-2.6.2-0-linux-x64-installer.run
chmod 755 chmod 755 bitnami-redmine-2.6.2-0-linux-x64-installer.run
sudo ./bitnami-redmine-2.6.2-0-linux-x64-installer.run
```

Here's the run down of the bitnami install script:

```
Language Selection

Please select the installation language
[1] English - English
[2] Spanish - Español
[3] Japanese - 日本語
[4] Korean - 한국어
[5] Simplified Chinese - 简体中文
[6] Hebrew - עברית
[7] German - Deutsch
[8] Romanian - Română
[9] Russian - Русский
Please choose an option [1] :
----------------------------------------------------------------------------
Welcome to the Bitnami Redmine Stack Setup Wizard.

----------------------------------------------------------------------------
Select the components you want to install; clear the components you do not want
to install. Click Next when you are ready to continue.

Subversion [Y/n] :

PhpMyAdmin [Y/n] :Y

Redmine : Y (Cannot be edited)

Git [Y/n] :Y

Is the selection above correct? [Y/n]: Y

----------------------------------------------------------------------------
Installation folder

Please, choose a folder to install Bitnami Redmine Stack

Select a folder [/opt/redmine-2.6.2-0]:

----------------------------------------------------------------------------
Create Admin account

Bitnami Redmine Stack admin user creation

Your real name [User Name]: Development

Email Address [user@example.com]: {scrubbed}

Login [user]: {scrubbed}

Password :
Please confirm your password :
----------------------------------------------------------------------------
Language for default data configuration

Select your language for default data configuration:

[1] Bulgarian
[2] Czech
[3] German
[4] English
[5] Spanish
[6] French
[7] Hebrew
[8] Italian
[9] Japanese
[10] Korean
[11] Dutch
[12] Polish
[13] Portuguese
[14] Portuguese/Brazilian
[15] Romanian
[16] Russian
[17] Serbian
[18] Swedish
[19] Chinese
[20] Chinese/Taiwan
Please choose an option [4] :

Do you want to configure mail support? [y/N]:

----------------------------------------------------------------------------
Setup is now ready to begin installing Bitnami Redmine Stack on your computer.

Do you want to continue? [Y/n]:
```

Since I'm on a HyperV vritual machine, I have made a checkpoint for the VM here. So I can return to this state and test any plugins or various versions there of. I am at this point running a vanilla Redmine 2.6.2 instance with no issues. Also important to note is that I am not running as root from this point on unless noted otherwise.

Installing Redmine_Jenkins:

```
sudo yum -y install git
cd /opt/redmine-2.6.2-0
sudo ./use_redmine
cd /opt/redmine-2.6.2-0/apps/redmine/htdocs/plugins
git clone https://github.com/jbox-web/redmine_bootstrap_kit.git
git clone https://github.com/jbox-web/redmine_jenkins.git
cd /opt/redmine-2.6.2-0/apps/redmine/htdocs
bundle install
```

I get the following errors:

Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for all non-root users on this machine.
You cannot specify the same gem twice with different version requirements.
You specified: shoulda-matchers (= 1.4.1) and shoulda-matchers (~> 2.7.0)

**Comment by webfractor** [comment]
Update, I redid the bitnami install as a regular user and was able to get through the install directions you have here:
https://github.com/jbox-web/redmine_jenkins/wiki/Step-by-step-installation-instructions

However, when I go into Redmine now, I only see the Redmine Bootstrap Kit, no Redmine_Jenkins

**Comment by webfractor** [comment]
Here's the full breakdown:

```
wget https://downloads.bitnami.com/files/stacks/redmine/2.6.2-0/bitnami-redmine-2.6.2-0-linux-x64-installer.run
chmod +x bitnami-redmine-2.6.2-0-linux-x64-installer.run
cd /opt/
sudo mkdir redmine
sudo chgrp developers redmine/
sudo chmod g+w redmine/
./bitnami-redmine-2.6.2-0-linux-x64-installer.run
```

Bitnami Installer:

```
Language Selection

Please select the installation language
[1] English - English
[2] Spanish - Español
[3] Japanese - 日本語
[4] Korean - 한국어
[5] Simplified Chinese - 简体中文
[6] Hebrew - עברית
[7] German - Deutsch
[8] Romanian - Română
[9] Russian - Русский
Please choose an option [1] :
----------------------------------------------------------------------------
Welcome to the Bitnami Redmine Stack Setup Wizard.

----------------------------------------------------------------------------
Select the components you want to install; clear the components you do not want
to install. Click Next when you are ready to continue.

Subversion [Y/n] :Y

PhpMyAdmin [Y/n] :Y

Redmine : Y (Cannot be edited)

Git [Y/n] :Y

Is the selection above correct? [Y/n]: Y

----------------------------------------------------------------------------
Installation folder

Please, choose a folder to install Bitnami Redmine Stack

Select a folder [/home/jgreen/redmine-2.6.2-0]: /opt/redmine

----------------------------------------------------------------------------
Create Admin account

Bitnami Redmine Stack admin user creation

Your real name [User Name]: Development

Email Address [user@example.com]: {scrubbed}

Login [user]: {scrubbed}

Password :
Please confirm your password :
----------------------------------------------------------------------------
Language for default data configuration

Select your language for default data configuration:

[1] Bulgarian
[2] Czech
[3] German
[4] English
[5] Spanish
[6] French
[7] Hebrew
[8] Italian
[9] Japanese
[10] Korean
[11] Dutch
[12] Polish
[13] Portuguese
[14] Portuguese/Brazilian
[15] Romanian
[16] Russian
[17] Serbian
[18] Swedish
[19] Chinese
[20] Chinese/Taiwan
Please choose an option [4] :

Do you want to configure mail support? [y/N]:

----------------------------------------------------------------------------
Setup is now ready to begin installing Bitnami Redmine Stack on your computer.

Do you want to continue? [Y/n]:

----------------------------------------------------------------------------
Please wait while Setup installs Bitnami Redmine Stack on your computer.

 Installing
 0% ______________ 50% ______________ 100%
 #########################################

----------------------------------------------------------------------------
Setup has finished installing Bitnami Redmine Stack on your computer.

Launch Redmine application. [Y/n]:
```

able to access redmine

redmine_jenkins install

```
cd /opt/redmine/apps/redmine/htdocs/plugins/
git clone https://github.com/jbox-web/redmine_bootstrap_kit.git
git clone https://github.com/jbox-web/redmine_jenkins.git
```

as a regular user:

```
bundle install
-bash: bundle: command not found
```

switching to redmine user

```
cd /opt/redmine
./use_redmine
cd /opt/redmine/apps/redmine/htdocs
 bundle install
```

output:

```
Using rake 10.4.2
Using i18n 0.6.11
Using multi_json 1.10.1
Using activesupport 3.2.21
Using builder 3.0.4
Using activemodel 3.2.21
Using erubis 2.7.0
Using journey 1.0.4
Using rack 1.4.5
Using rack-cache 1.2
Using rack-test 0.6.3
Using hike 1.2.3
Using tilt 1.4.1
Using sprockets 2.2.3
Using actionpack 3.2.21
Using mime-types 1.25.1
Using polyglot 0.3.5
Using treetop 1.4.15
Using mail 2.5.4
Using actionmailer 3.2.21
Using arel 3.0.3
Using tzinfo 0.3.43
Using activerecord 3.2.21
Using activeresource 3.2.21
Using coderay 1.1.0
Using rack-ssl 1.3.4
Using json 1.8.2
Using rdoc 3.12.2
Using thor 0.19.1
Using railties 3.2.21
Using jquery-rails 3.1.2
Using mysql2 0.3.14
Using net-ldap 0.3.1
Using ruby-openid 2.3.0
Using rack-openid 1.4.2
Using bundler 1.7.0
Using rails 3.2.21
Using rbpdf 1.18.5
Using redcarpet 2.3.0
Using request_store 1.0.5
Using rmagick 2.13.4
Your bundle is complete!
Gems in the groups development, test, sqlite, test and sqlite were not installed.
It was installed into ./vendor/bundle
```

finally:

```
bundle exec rake redmine:plugins:migrate RAILS_ENV=production
./ctlscript.sh restart
```

![redmine_jenkins_missing](https://cloud.githubusercontent.com/assets/4032926/6622808/f72519b4-c8ac-11e4-822b-eec00087c877.png)

**Comment by webfractor** [comment]
Logs:

/opt/redmine/apps/redmine/htdocs/log/production.log:

```
Connecting to database specified by database.yml
Creating scope :system. Overwriting existing method Enumeration.system.
Connecting to database specified by database.yml
Creating scope :system. Overwriting existing method Enumeration.system.
Creating scope :sorted. Overwriting existing method User.sorted.
Creating scope :sorted. Overwriting existing method Group.sorted.
Started GET "/redmine" for 127.0.0.1 at 2015-03-12 10:54:45 -0500
Started GET "/redmine" for 127.0.0.1 at 2015-03-12 10:54:45 -0500
Processing by WelcomeController#index as HTML
Processing by WelcomeController#index as HTML
Completed 500 Internal Server Error in 587.7ms

RuntimeError (Unable to create the anonymous user.):
  app/models/user.rb:655:in `anonymous'
  app/models/user.rb:646:in `current'
  app/controllers/application_controller.rb:103:in `user_setup'


  Current user: anonymous
  Rendered welcome/index.html.erb within layouts/base (71.5ms)
Completed 200 OK in 1571.6ms (Views: 345.6ms | ActiveRecord: 292.6ms)
Started GET "/redmine/" for 10.101.120.30 at 2015-03-12 11:10:38 -0500
Processing by WelcomeController#index as HTML
  Current user: anonymous
  Rendered welcome/index.html.erb within layouts/base (39.0ms)
Completed 200 OK in 293.3ms (Views: 66.2ms | ActiveRecord: 8.4ms)
Connecting to database specified by database.yml
Connecting to database specified by database.yml
Creating scope :system. Overwriting existing method Enumeration.system.
Creating scope :sorted. Overwriting existing method User.sorted.
Creating scope :sorted. Overwriting existing method Group.sorted.
Started GET "/redmine" for 127.0.0.1 at 2015-03-12 11:19:53 -0500
Started GET "/redmine" for 127.0.0.1 at 2015-03-12 11:19:53 -0500
Processing by WelcomeController#index as HTML
Processing by WelcomeController#index as HTML
  Current user: anonymous
  Current user: anonymous
  Rendered welcome/index.html.erb within layouts/base (96.7ms)
  Rendered welcome/index.html.erb within layouts/base (108.2ms)
Completed 200 OK in 1064.8ms (Views: 194.2ms | ActiveRecord: 95.7ms)
Completed 200 OK in 1074.5ms (Views: 198.8ms | ActiveRecord: 83.5ms)
Started GET "/redmine/" for 10.101.120.30 at 2015-03-12 11:20:00 -0500
Processing by WelcomeController#index as HTML
  Current user: anonymous
  Rendered welcome/index.html.erb within layouts/base (1.2ms)
Completed 200 OK in 15.4ms (Views: 6.8ms | ActiveRecord: 1.8ms)
Started GET "/redmine/login" for 10.101.120.30 at 2015-03-12 11:20:02 -0500
Processing by AccountController#login as HTML
  Current user: anonymous
  Rendered account/login.html.erb within layouts/base (5.7ms)
Completed 200 OK in 21.7ms (Views: 17.0ms | ActiveRecord: 1.4ms)
Started POST "/redmine/login" for 10.101.120.30 at 2015-03-12 11:20:08 -0500
Processing by AccountController#login as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"Cvoe5ZYcMami2ZVKZQft3JO7g6DG534h9HVlOMr5LjI=", "back_url"=>"http://redmine.lab.dsrglobal.com:8080/redmine/", "username"=>"devadm", "password"=>"[FILTERED]", "login"=>"Login »"}
  Current user: anonymous
Successful authentication for 'devadm' from 10.101.120.30 at 2015-03-12 16:20:08 UTC
Redirected to http://redmine.lab.dsrglobal.com:8080/redmine/
Completed 302 Found in 18.1ms (ActiveRecord: 10.3ms)
Started GET "/redmine/" for 10.101.120.30 at 2015-03-12 11:20:08 -0500
Processing by WelcomeController#index as HTML
  Current user: devadm (id=1)
  Rendered welcome/index.html.erb within layouts/base (99.6ms)
Completed 200 OK in 195.7ms (Views: 145.3ms | ActiveRecord: 39.8ms)
Started GET "/redmine/admin" for 10.101.120.30 at 2015-03-12 11:20:12 -0500
Processing by AdminController#index as HTML
  Current user: devadm (id=1)
  Rendered admin/_menu.html.erb (8.1ms)
  Rendered admin/index.html.erb within layouts/admin (22.6ms)
  Rendered layouts/base.html.erb (12.1ms)
Completed 200 OK in 53.7ms (Views: 47.3ms | ActiveRecord: 2.0ms)
Started GET "/redmine/admin/plugins" for 10.101.120.30 at 2015-03-12 11:20:15 -0500
Processing by AdminController#plugins as HTML
  Current user: devadm (id=1)
  Rendered admin/plugins.html.erb within layouts/admin (2.4ms)
  Rendered admin/_menu.html.erb (4.3ms)
  Rendered layouts/base.html.erb (10.8ms)
Completed 200 OK in 32.3ms (Views: 26.3ms | ActiveRecord: 1.7ms)
Started GET "/redmine/admin/plugins" for 10.101.120.30 at 2015-03-12 11:31:35 -0500
Processing by AdminController#plugins as HTML
  Current user: devadm (id=1)
  Rendered admin/plugins.html.erb within layouts/admin (0.7ms)
  Rendered admin/_menu.html.erb (3.2ms)
  Rendered layouts/base.html.erb (8.1ms)
Completed 200 OK in 16.2ms (Views: 12.1ms | ActiveRecord: 1.6ms)
Started GET "/redmine/admin/plugins" for 10.101.120.30 at 2015-03-12 11:31:36 -0500
Processing by AdminController#plugins as HTML
  Current user: devadm (id=1)
  Rendered admin/plugins.html.erb within layouts/admin (0.8ms)
  Rendered admin/_menu.html.erb (3.5ms)
  Rendered layouts/base.html.erb (8.4ms)
Completed 200 OK in 16.2ms (Views: 12.7ms | ActiveRecord: 1.4ms)
Started GET "/redmine/settings" for 10.101.120.30 at 2015-03-12 11:31:40 -0500
Processing by SettingsController#index as HTML
  Current user: devadm (id=1)
  Rendered settings/_general.html.erb (24.5ms)
  Rendered settings/_display.html.erb (1878.2ms)
  Rendered settings/_authentication.html.erb (8.0ms)
  Rendered settings/_projects.html.erb (20.1ms)
  Rendered queries/_columns.html.erb (15.3ms)
  Rendered settings/_issues.html.erb (162.0ms)
  Rendered settings/_notifications.html.erb (11.7ms)
  Rendered settings/_mail_handler.html.erb (5.5ms)
  Rendered settings/_repositories.html.erb (400.0ms)
  Rendered common/_tabs.html.erb (2547.2ms)
  Rendered settings/edit.html.erb within layouts/admin (2557.3ms)
Rendered admin/_menu.html.erb (3.9ms)
  Rendered layouts/base.html.erb (9.7ms)
Completed 200 OK in 2757.9ms (Views: 2551.9ms | ActiveRecord: 32.6ms)
Started POST "/redmine/settings/edit" for 10.101.120.30 at 2015-03-12 11:35:07 -0500
Processing by SettingsController#edit as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"RKcB7AkYfjOnjW5efHx+bmXs9woR8FDG4lNQnOQLbsI=", "settings"=>{"app_title"=>"Redmine", "welcome_text"=>"", "attachment_max_size"=>"5120", "per_page_options"=>"25,50,100", "activity_days_default"=>"30", "host_name"=>"redmine.lab.dsrglobal.com:8080/redmine", "protocol"=>"http", "text_formatting"=>"textile", "cache_formatted_text"=>"0", "wiki_compression"=>"", "feeds_limit"=>"15", "file_max_size_displayed"=>"512", "diff_max_lines_displayed"=>"1500", "repositories_encodings"=>""}, "commit"=>"Save"}
  Current user: devadm (id=1)
Redirected to http://redmine.lab.dsrglobal.com:8080/redmine/settings
Completed 302 Found in 80.4ms (ActiveRecord: 47.5ms)
Started GET "/redmine/settings" for 10.101.120.30 at 2015-03-12 11:35:07 -0500
Processing by SettingsController#index as HTML
Settings cache cleared.
  Current user: devadm (id=1)
  Rendered settings/_general.html.erb (108.1ms)
  Rendered settings/_display.html.erb (10.8ms)
  Rendered settings/_authentication.html.erb (11.2ms)
  Rendered settings/_projects.html.erb (9.5ms)
  Rendered queries/_columns.html.erb (3.6ms)
  Rendered settings/_issues.html.erb (24.3ms)
  Rendered settings/_notifications.html.erb (18.0ms)
  Rendered settings/_mail_handler.html.erb (3.9ms)
  Rendered settings/_repositories.html.erb (14.6ms)
  Rendered common/_tabs.html.erb (204.6ms)
  Rendered settings/edit.html.erb within layouts/admin (205.0ms)
  Rendered admin/_menu.html.erb (3.5ms)
  Rendered layouts/base.html.erb (8.8ms)
Completed 200 OK in 223.7ms (Views: 186.0ms | ActiveRecord: 33.1ms)
Started GET "/redmine/admin/plugins" for 10.101.120.30 at 2015-03-12 11:35:21 -0500
Processing by AdminController#plugins as HTML
  Current user: devadm (id=1)
  Rendered admin/plugins.html.erb within layouts/admin (1.0ms)
  Rendered admin/_menu.html.erb (3.6ms)
  Rendered layouts/base.html.erb (9.4ms)
Completed 200 OK in 18.6ms (Views: 14.2ms | ActiveRecord: 1.8ms)
Started GET "/redmine/settings" for 10.101.120.30 at 2015-03-12 11:35:23 -0500
Processing by SettingsController#index as HTML
  Current user: devadm (id=1)
  Rendered settings/_general.html.erb (5.9ms)
  Rendered settings/_display.html.erb (5.6ms)
  Rendered settings/_authentication.html.erb (4.6ms)
  Rendered settings/_projects.html.erb (7.7ms)
  Rendered queries/_columns.html.erb (3.8ms)
  Rendered settings/_issues.html.erb (8.9ms)
  Rendered settings/_notifications.html.erb (5.0ms)
  Rendered settings/_mail_handler.html.erb (1.7ms)
  Rendered settings/_repositories.html.erb (9.8ms)
  Rendered common/_tabs.html.erb (53.4ms)
  Rendered settings/edit.html.erb within layouts/admin (54.0ms)
  Rendered admin/_menu.html.erb (3.7ms)
  Rendered layouts/base.html.erb (9.4ms)
Completed 200 OK in 72.3ms (Views: 65.9ms | ActiveRecord: 3.1ms)
Started GET "/redmine/admin/plugins" for 10.101.120.30 at 2015-03-12 11:36:05 -0500
Processing by AdminController#plugins as HTML
  Current user: devadm (id=1)
  Rendered admin/plugins.html.erb within layouts/admin (1.0ms)
  Rendered admin/_menu.html.erb (3.5ms)
  Rendered layouts/base.html.erb (8.5ms)
Completed 200 OK in 17.2ms (Views: 13.1ms | ActiveRecord: 1.6ms)
```

**Comment by webfractor** [comment]
Just for completeness, I rolled back to vanilla redmine, no plugins and did the following:

```
cd /opt/redmine/apps/redmine/htdocs/plugins/
git clone https://github.com/jbox-web/redmine_bootstrap_kit.git
git clone https://github.com/jbox-web/redmine_jenkins.git
cd /opt/redmine
./use_redmine
cd /opt/redmine/apps/redmine/htdocs
 bundle install --deployment
```

And got the following:

```
You cannot specify the same gem twice with different version requirements.
You specified: shoulda-matchers (= 1.4.1) and shoulda-matchers (~> 2.7.0)
```

**Comment by n-rodriguez** [comment]
> You specified: shoulda-matchers (= 1.4.1) and shoulda-matchers (~> 2.7.0)

Comment the development group in Jenkins plugin Gemfile.

**Comment by webfractor** [comment]
Commented out the entire section

```
source 'https://rubygems.org'

gem 'jenkins_api_client', '~> 1.3.0'
gem 'will_paginate'

# HAML views
gem 'haml-rails'

#group :development, :test do
#  gem 'rspec', '~> 3.0.0'
#  gem 'rspec-rails', '~> 3.0.1'

#  gem 'shoulda', '~> 3.5.0'
#  gem 'shoulda-matchers', '~> 2.7.0'
#  gem 'shoulda-context'

#  gem 'factory_girl'
#  gem 'factory_girl_rails'
#  gem 'faker'
#  gem 'database_cleaner'

  # Code coverage
#  gem 'simplecov', '~> 0.9.1'
#  gem 'simplecov-rcov'

  # Junit results
#  gem 'ci_reporter_rspec', '~> 1.0.0'
#end

```

Did the following:

```
 cd /opt/redmine
./use_redmine
cd /opt/redmine/apps/redmine/htdocs
bundle install --deployment
```

Results in:

```
You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

You have added to the Gemfile:
* jenkins_api_client (~> 1.3.0)
* will_paginate
* haml-rails
```

tried:

```
mv Gemfile.lock Gemfile.lock.orig
bundle install --deployment
```

Got:

```
The --deployment flag requires a Gemfile.lock. Please make sure you have checked your Gemfile.lock into version control before deploying.
```

Tried:

```
bundle install
```

Got :

```
You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

If this is a development machine, remove the Gemfile freeze
by running `bundle install --no-deployment`.

You have added to the Gemfile:
* source: rubygems repository https://rubygems.org/
* rails (= 3.2.21)
* jquery-rails (~> 3.1.1)
* coderay (~> 1.1.0)
* fastercsv (~> 1.5.0)
* builder (>= 3.0.4)
* request_store (= 1.0.5)
* mime-types
* rbpdf (~> 1.18.5)
* i18n (~> 0.6.11)
* net-ldap (~> 0.3.1)
* ruby-openid (~> 2.3.0)
* rack-openid
* rmagick (>= 2.0.0)
* redcarpet (~> 2.3.0)
* activerecord-jdbc-adapter (~> 1.3.2)
* mysql2 (= 0.3.14)
* activerecord-jdbcmysql-adapter
* rdoc (>= 2.4.2)
* yard
* shoulda-matchers (= 1.4.1)
* mocha (~> 1.0.0)
* capybara
* selenium-webdriver
* jenkins_api_client (~> 1.3.0)
* will_paginate
* haml-rails
```

Did some googling and found this might have to do with the gems install in vendor...

```
cd /opt/redmine/apps/redmine/htdocs/vendor/bundle/ruby/2.0.0
cp -r gems/ gems.old/
cd gems
rm -rf *
cd /opt/redmine/apps/redmine/htdocs/
bundle install
```

Got the same error.

I have reverted to the state I was in before a began the step in this bug. Then I repeated the commenting.

I know that a lot of this process is me not knowing much about ruby, rails, or how this stuff is managed. I really appreciate you hanging in here with me.

Thanks

**Comment by webfractor** [comment]
It occurred to me perhaps I should run that in the plugin directory:

```
cd /opt/redmine/apps/redmine/htdocs/plugins/redmine_jenkins
bundle install
```

Got:

```
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Installing rake 10.4.2
Using i18n 0.7.0

Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native extension.

    /opt/redmine/ruby/bin/ruby extconf.rb
creating Makefile

make "DESTDIR="
compiling generator.c
make: gcc: Command not found
make: *** [generator.o] Error 127


Gem files will remain installed in /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/json-1.8.2 for inspection.
Results logged to /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/json-1.8.2/ext/json/ext/generator/gem_make.out
An error occurred while installing json (1.8.2), and Bundler cannot continue.
Make sure that `gem install json -v '1.8.2'` succeeds before bundling.
```

ran

```
cat /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/json-1.8.2/ext/json/ext/generator/gem_make.out

/opt/redmine/ruby/bin/ruby extconf.rb
creating Makefile

make "DESTDIR="
compiling generator.c
make: gcc: Command not found
make: *** [generator.o] Error 127
```

installed gcc

```
 bundle install
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Using rake 10.4.2
Using i18n 0.7.0
Installing json 1.8.2
Installing minitest 5.5.1
Installing thread_safe 0.3.5
Installing tzinfo 1.2.2
Using activesupport 4.2.0
Installing builder 3.2.2
Using erubis 2.7.0
Installing mini_portile 0.6.2

Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native extension.

    /opt/redmine/ruby/bin/ruby extconf.rb --with-opt-dir=/opt/redmine/common
checking if the C compiler accepts -I/opt/redmine/common/include/ImageMagick -I/opt/redmine/common/include -I/opt/redmine/common/include/ImageMagick -I/opt/redmine/common/include ... yes
Building nokogiri using packaged libraries.
checking for gzdopen() in -lz... yes
checking for iconv using --with-opt-* flags... yes
************************************************************************
IMPORTANT NOTICE:

Building Nokogiri with a packaged version of libxml2-2.9.2
with the following patches applied:
        - 0001-Revert-Missing-initialization-for-the-catalog-module.patch
        - 0002-Fix-missing-entities-after-CVE-2014-3660-fix.patch

Team Nokogiri will keep on doing their best to provide security
updates in a timely manner, but if this is a concern for you and want
to use the system library instead; abort this installation process and
reinstall nokogiri as follows:

    gem install nokogiri -- --use-system-libraries
        [--with-xml2-config=/path/to/xml2-config]
        [--with-xslt-config=/path/to/xslt-config]

If you are using Bundler, tell it to use the option:

    bundle config build.nokogiri --use-system-libraries
    bundle install

Note, however, that nokogiri is not fully compatible with arbitrary
versions of libxml2 provided by OS/package vendors.
************************************************************************
Extracting libxml2-2.9.2.tar.gz into tmp/x86_64-unknown-linux-gnu/ports/libxml2/2.9.2... OK
Running patch with /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/nokogiri-1.6.6.2/ports/patches/libxml2/0001-Revert-Missing-initialization-for-the-catalog-module.patch...
Running 'patch' for libxml2 2.9.2... ERROR, review '/opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/nokogiri-1.6.6.2/ext/nokogiri/tmp/x86_64-unknown-linux-gnu/ports/libxml2/2.9.2/patch.log' to see what happened.
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=/opt/redmine/ruby/bin/ruby
        --help
        --clean
        --use-system-libraries
        --enable-static
        --disable-static
        --with-zlib-dir
        --without-zlib-dir
        --with-zlib-include
        --without-zlib-include=${zlib-dir}/include
        --with-zlib-lib
        --without-zlib-lib=${zlib-dir}/lib
        --enable-cross-build
        --disable-cross-build
/opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/mini_portile-0.6.2/lib/mini_portile.rb:279:in `block in execute': Failed to complete patch task (RuntimeError)
        from /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/mini_portile-0.6.2/lib/mini_portile.rb:271:in `chdir'
        from /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/mini_portile-0.6.2/lib/mini_portile.rb:271:in `execute'
        from extconf.rb:311:in `block in patch'
        from extconf.rb:308:in `each'
        from extconf.rb:308:in `patch'
        from /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/mini_portile-0.6.2/lib/mini_portile.rb:108:in `cook'
        from extconf.rb:278:in `block in process_recipe'
        from extconf.rb:177:in `tap'
        from extconf.rb:177:in `process_recipe'
        from extconf.rb:475:in `<main>'


Gem files will remain installed in /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/nokogiri-1.6.6.2 for inspection.
Results logged to /opt/redmine/ruby/lib/ruby/gems/2.0.0/gems/nokogiri-1.6.6.2/ext/nokogiri/gem_make.out
An error occurred while installing nokogiri (1.6.6.2), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.6.6.2'` succeeds before bundling.
```

tried

```
bundle config build.nokogiri --use-system-libraries
bundle install


Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Using rake 10.4.2
Using i18n 0.7.0
Using json 1.8.2
Using minitest 5.5.1
Using thread_safe 0.3.5
Using tzinfo 1.2.2
Using activesupport 4.2.0
Using builder 3.2.2
Using erubis 2.7.0
Using mini_portile 0.6.2
Installing nokogiri 1.6.6.2
Using rails-deprecated_sanitizer 1.0.3
Using rails-dom-testing 1.0.5
Using loofah 2.0.1
Installing rails-html-sanitizer 1.0.2
Using actionview 4.2.0
Using rack 1.6.0
Installing rack-test 0.6.3
Using actionpack 4.2.0
Installing tilt 2.0.1
Installing haml 4.0.6
Installing sexp_processor 4.5.0
Installing ruby_parser 3.6.5
Installing html2haml 2.0.0
Installing thor 0.19.1
Using railties 4.2.0
Installing haml-rails 0.9.0
Installing mixlib-shellout 2.0.1
Installing terminal-table 1.4.5
Installing jenkins_api_client 1.3.0
Installing will_paginate 3.0.7
Using bundler 1.7.0
Your bundle is complete!
Use `bundle show [gemname]` to see where a bundled gem is installed.
Post-install message from haml:

HEADS UP! Haml 4.0 has many improvements, but also has changes that may break
your application:

* Support for Ruby 1.8.6 dropped
* Support for Rails 2 dropped
* Sass filter now always outputs <style> tags
* Data attributes are now hyphenated, not underscored
* html2haml utility moved to the html2haml gem
* Textile and Maruku filters moved to the haml-contrib gem

For more info see:

http://rubydoc.info/github/haml/haml/file/CHANGELOG.md
```

then tried

```
cd /opt/redmine/apps/redmine/htdocs/
bundle install --deployment
You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

You have added to the Gemfile:
* jenkins_api_client (~> 1.3.0)
* will_paginate
* haml-rails
```

```
cd /opt/redmine/apps/redmine/htdocs/plugins/redmine_jenkins/
bundle install --deployment


Fetching gem metadata from https://rubygems.org/..........
Installing rake 10.4.2
Installing i18n 0.7.0
Installing json 1.8.2
Installing minitest 5.5.1
Installing thread_safe 0.3.5
Installing tzinfo 1.2.2
Installing activesupport 4.2.0
Installing builder 3.2.2
Installing erubis 2.7.0
Installing mini_portile 0.6.2
Installing nokogiri 1.6.6.2
Installing rails-deprecated_sanitizer 1.0.3
Installing rails-dom-testing 1.0.5
Installing loofah 2.0.1
Installing rails-html-sanitizer 1.0.2
Installing actionview 4.2.0
Installing rack 1.6.0
Installing rack-test 0.6.3
Installing actionpack 4.2.0
Installing tilt 2.0.1
Installing haml 4.0.6
Installing sexp_processor 4.5.0
Installing ruby_parser 3.6.5
Installing html2haml 2.0.0
Installing thor 0.19.1
Installing railties 4.2.0
Installing haml-rails 0.9.0
Installing mixlib-shellout 2.0.1
Installing terminal-table 1.4.5
Installing jenkins_api_client 1.3.0
Installing will_paginate 3.0.7
Using bundler 1.7.0
Your bundle is complete!
It was installed into ./vendor/bundle
Post-install message from haml:

HEADS UP! Haml 4.0 has many improvements, but also has changes that may break
your application:

* Support for Ruby 1.8.6 dropped
* Support for Rails 2 dropped
* Sass filter now always outputs <style> tags
* Data attributes are now hyphenated, not underscored
* html2haml utility moved to the html2haml gem
* Textile and Maruku filters moved to the haml-contrib gem

For more info see:

http://rubydoc.info/github/haml/haml/file/CHANGELOG.md
```

```
cd /opt/redmine/apps/redmine/htdocs/
bundle install --deployment
You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

You have added to the Gemfile:
* jenkins_api_client (~> 1.3.0)
* will_paginate
* haml-rails
```

Reverted again, waiting on your guidance now

**Comment by n-rodriguez** [comment]
This is a Bitnami Stack thing... Need to be investigated...

**Comment by webfractor** [comment]
Ok, I think I not only got this installed, but it's actually working now!

Here's the steps I had to take starting from right after the bitnami install:

```
sudo yum -y install git gcc
 cd /opt/redmine/apps/redmine/htdocs/plugins/
git clone https://github.com/jbox-web/redmine_bootstrap_kit.git
git clone https://github.com/jbox-web/redmine_jenkins.git
vim redmine_jenkins/Gemfile

#Comment out the entire development group
cd /opt/redmine/
./use_redmine
cd /opt/redmine/apps/redmine/htdocs/plugins/redmine_jenkins
bundle config build.nokogiri --use-system-libraries
bundle install
cd /opt/redmine/apps/redmine/htdocs/
bundle update
bundle install -no-deployment
bundle exec rake redmine:plugins:migrate RAILS_ENV=production
cd /opt/redmine/
./ctlscript.sh restart
```

I now see both plugins in the plugin page:

![jenkins_redmine_plugin](https://cloud.githubusercontent.com/assets/4032926/6670331/fb2e64ee-cbcb-11e4-8e9e-c993eec41d68.png)

I have the option for jenkins in a new project. Configuration test passed by showing me how many jobs there are, and i was able to attach a job to the project.

Thanks for your help!

**Comment by webfractor** [comment]
I think you should add this to your wiki, I'm sure I'm not the only that is having issues and using bitnami. I can provide a condenced howto after I proof my steps and build this server out from scratch.

**Comment by n-rodriguez** [comment]
> I think you should add this to your wiki

I will!

> I'm sure I'm not the only that is having issues and using bitnami.

I confirm, you're not alone : https://github.com/jbox-web/redmine_git_hosting/issues/376 http://jbox-web.github.io/redmine_git_hosting/guide/bitnami-installation/

> I can provide a condenced howto after I proof my steps and build this server out from scratch.

That would be great !

Thank you!

**Comment by webfractor** [comment]
So, I went back through this working from scratch to proof the install and apparently BitNami jerks the redmine config around just enough that I can't seem to get this workin until I install as a user. Install as root just leads to more and more issues. So I gave up.

I have now installed redmine from scratch and run into a new issue in another ticket...

**Comment by n-rodriguez** [comment]
> I have now installed redmine from scratch and run into a new issue in another ticket...

which one?

**Comment by webfractor** [comment]
It was issue #36.

I have abandoned bitnami. I think most of my issues came from configurations that bitnami used versus issues with your plugin. I just could not get bitnami setup on the server the way I wanted. I am now running Redmine installed from tarballs and not having any issues since you 1.0.1 release.
